File: DocMgmtUi/docmgmtui/src/types/Customer.ts
import { ControllerRenderProps } from "react-hook-form";

export type Customer = {
    id?: string,
    firstName: string;
    lastName: string;
    email: string;
    address: string;
    city: string;
    state: string;
    postalCode: string;
    country: string;
    phone: string;
    gender: string;
    dateCreated?: Date;
    documents: string[];
};

// export type Gender = ControllerRenderProps<Entity, "gender">

export const emptyCustomer = {
    firstName: 'David',
    lastName: 'Baker',
    email: 'dbaker@gmail.com',
    address: '25 Wall Street',
    city: 'Halifax',
    state: 'NS',
    postalCode: 'B3A 2K9',
    country: 'Canada',
    phone: '9034854844',
    gender: 'Male',
}

File: DocMgmtApi/DocMgmtApi/Models/Customer.cs
namespace DocMgmtApi.Models
{
    public class Customer
    {
        public Guid Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Email { get; set; }
        public string Address { get; set; }
        public string City { get; set; }
        public string State { get; set; }
        public string PostalCode { get; set; }
        public string Country { get; set; }
        public string Phone { get; set; }
        public string Gender { get; set; }
        public DateTime DateCreated { get; set; } = DateTime.Now;
        public virtual ICollection<Document> Documents { get; set; } = new List<Document>();
    }
}



